### PR TITLE
Feat/data 3498 switch sampling queries

### PR DESCRIPTION
**Pull request by KatzperWoz**
Tl;dr: Switch sampling queries, instead of querying sampled data we first create a transient table with sampled data and then fetch transient table to local memory. On top of that some code refactors.

Context:
The goal is to transition from storing temporary data in memory to storing it in Snowflake. 

During data extraction, Snowshu should utilize a previously created function to generate schemas and tables in Snowflake. After creating these schemas and tables, it should transfer data from the temporary tables into the relation.data attribute.

After processing the entire connected subgraph, the temporary schemas and tables in Snowflake should be cleaned up. 

The core part responsible for it is placed in ```graph_set_runner.py``` module.

On top of that there are some refactors making code easier to read and understand.

Test Plan:
If someone would like to test it, I'd recommend to install this package from source and later try to execute snowshu create function on one of snowflake repos eg. hu-dw-artifact, hu-dw-datawarehouse

### CODE REVIEW

**Review by Gal40n04ek — APPROVED** [review]

**Review by glysav — APPROVED** [review]
This changes looks good to me. Works fine locally!
